### PR TITLE
Legacy core with navigator

### PR DESCRIPTION
**Pull request by HamdanSCode**
Implement the navigator in FA-104

### CODE REVIEW

File: applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyCreateAccountViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    private fun createAccountNormalView() {
+        errorId = null
+        updateCreateAccountState()
+    }
+
+    fun createAccountDismissErrorClicked() {
+        createAccountNormalView()
+    }
+
+    fun createAccount() {
```
Comment (ForzenAcademy) [inline applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyCreateAccountViewModel.kt]: Things like this function make me think an abstract ViewModel base class is needed so there's no duplication

File: applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyCreateAccountViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    private var errorId: Int? = null
+    private var firstName: Entry = Entry("", LoginError.NameError.None)
+    private var lastName: Entry = Entry("", LoginError.NameError.None)
+    private var birthDay: Entry = Entry("", LoginError.BirthDateError.None)
+    private var email: Entry = Entry("", LoginError.EmailError.None)
+    private var location: Entry = Entry("", LoginError.LocationError.None)
+    private var isDateDialogOpen: Boolean = false
+    private var isLoading: Boolean = false
+
+    var onCreateAccountUpdate: ((CreateAccountState) -> Unit)? = null
```
Comment (ForzenAcademy) [inline applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyCreateAccountViewModel.kt]: Stateflow

File: applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyLoginViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    private var email: Entry = Entry("", LoginError.EmailError.None)
+    private var code: Entry = Entry("", LoginError.CodeError.None)
+    private var showInfoDialog: Boolean = false
+    private var inputtingCode: Boolean = false
+    private var isLoading: Boolean = false
+    private var hasError: Boolean = false
+
+    var onLoginUpdate: ((LoginState) -> Unit)? = null
```
Comment (ForzenAcademy) [inline applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyLoginViewModel.kt]: Stateflow

File: applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyLoginViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+                code = tempCode
+                inputtingCode = true
+                isLoading = false
+                hasError = true
+            }
+            updateLoginState()
+        }
+    }
+
+    private fun requestLoginValidationCode() {
```
Comment (ForzenAcademy) [inline applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyLoginViewModel.kt]: Make a base viewmodel class

File: applegacy/src/main/res/values-night/themes.xml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<resources xmlns:tools="http://schemas.android.com/tools">
+    <!-- Base application theme. -->
+    <style name="Theme.Forzenbook" parent="Theme.MaterialComponents.DayNight.DarkActionBar">
+        <item name="legacy_background">@color/lion_yellow</item>
+
+        <item name="legacy_primary">@color/black</item>
```
Comment (ForzenAcademy) [inline applegacy/src/main/res/values-night/themes.xml]: camelCase, lots of comments from the other PR I think?

File: legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/NavigatorImpl.kt [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package com.hamdan.forzenbook.legacy.core.view
+
+class NavigatorImpl(
+    val routeToLogin: () -> Unit,
```
Comment (ForzenAcademy) [inline legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/NavigatorImpl.kt]: This IS the impl. An impl that takes in lambdas with the true impl doesn't make sense lol it's fine for this class to take in a context in its methods and route by itself

File: applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyCreateAccountViewModel.kt (hunk 1 above)
Comment (HamdanSCode) [reply applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyCreateAccountViewModel.kt]: I don't understand what you mean. what duplication is there in this context. Are you refering to between the two viewmodels. Are you refering to this createAccount and the createAccount in the app Viewmodel?

File: applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyCreateAccountViewModel.kt (hunk 2 above)
Comment (HamdanSCode) [reply applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyCreateAccountViewModel.kt]: turned into a stateflow

File: legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/NavigatorImpl.kt (hunk 6 above)
Comment (HamdanSCode) [reply legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/NavigatorImpl.kt]: ok that makes sense, I hope that's the only last step in that implementation concept that I'm missing considering how the struggle was trying to understand implementing the concept.

File: applegacy/src/main/res/values-night/themes.xml (hunk 5 above)
Comment (HamdanSCode) [reply applegacy/src/main/res/values-night/themes.xml]: I think I converted all snake case to camelcase, the only one that I did not was the colors which I left as for example dark_gray

File: applegacy/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <string name="app_name">appLegacy</string>
```
Comment (ForzenAcademy) [inline applegacy/src/main/res/values/strings.xml:2]: Use the app name here? Or update the manifest to not use this and use the app name string?

File: legacyCore/build.gradle [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+    }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
```
Comment (ForzenAcademy) [inline legacyCore/build.gradle:27]: Java 11

File: legacyCore/build.gradle [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
```
Comment (ForzenAcademy) [inline legacyCore/build.gradle:31]: Java 11

File: legacyCore/src/main/AndroidManifest.xml [hunk 10]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (ForzenAcademy) [inline legacyCore/src/main/AndroidManifest.xml:2]: Condense

File: legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/NavigatorImpl.kt [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+import android.content.Context
+
+class NavigatorImpl(
+    val legacyLoginContext: Context,
+    val legacyCreateAccountContext: Context,
+) : Navigator {
+    // Todo implement both when FA-100 and 102 are implemented
+    override fun navigateToLogin() {
+// may want to change this from implicit intent incase we want to fill in email other options are possible too
```
Comment (ForzenAcademy) [inline legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/NavigatorImpl.kt:11]: Any sort of implicit sharing would definitely be new and different functionality

File: legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/NavigatorImpl.kt [hunk 12]
```diff
@@ -0,0 +1,23 @@
+package com.hamdan.forzenbook.legacy.core.view
+
+import android.content.Context
+
+class NavigatorImpl(
+    val legacyLoginContext: Context,
```
Comment (ForzenAcademy) [inline legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/NavigatorImpl.kt:6]: Bring the context in the _methods_, not the constructor. Then you don't have to hold the context and be worried about leaks. Other than that your methods commented code looks about right

**Review by ForzenAcademy — APPROVED** [review]
Fix the things I pointed out and then merge

File: applegacy/src/main/res/values/strings.xml (hunk 7 above)
Comment (HamdanSCode) [reply applegacy/src/main/res/values/strings.xml:2]: These values files will be removed in the favour of the actual relevant value files, I'll leave a todo to remove them, for now it is updated to be Forzenbook

File: legacyCore/build.gradle (hunk 8 above)
Comment (HamdanSCode) [reply legacyCore/build.gradle:27]: thought I had these updated, updated to 11

File: legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/NavigatorImpl.kt (hunk 12 above)
Comment (HamdanSCode) [reply legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/NavigatorImpl.kt:6]: updated

File: legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/NavigatorImpl.kt (hunk 11 above)
Comment (HamdanSCode) [reply legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/NavigatorImpl.kt:11]: haha, mixed up explicit and implicit

**Pull request merged**